lib/handleRequest.js

```diff
@@ -82,6 +82,12 @@ function preValidationCallback (err, request, reply) {
 
   var result = validateSchema(reply.context, request)
   if (result) {
+    if (result.formatterError) {
+      result.message = result.formatterError
+      reply.code(500).send(result)
```

Comment:
```suggestion
      reply.code(400).send(result)
```

Here isn't  400?

Reply:
@Eomm requested that errors thrown inside `schemaErrorFormatter` will return `500`

Reply:
@Arnovsky what was the reason for that?

Reply:
@kibertoad Don't think it makes sense to return bad request in this case, since this is not a user based error, 500 implies that this is a fault in the server.

Reply:
When do we throw this error? Invalid schema definition on the server?

Reply:
When the `schemaErrorFormatter` passed throws an error, when it fails to format the error.

Reply:
Does it make sense to return that error to the user and not log server-side, if it has nothing to do with user?

Reply:
I am understanding correctly the code, I also vote for not replying with the error. For the same reason, I agree that it is a 500 error.

Reply:
I agree too. When I read the code, this part sounds like a `validationError` but is a runtime error.

Reply:
@kibertoad  @fox1t  @RafaelGSS What should be returned instead? Arguably no error should be returned to the user yet there are many examples I found in the code that do the same, What if we throw inside a route? it will return a '500' error with the message, how is this any different?

Reply:
@Arnovsky Let us imagine the real world scenario here. Imagine that I am Netflix API developer, who wrote fastify app and had made error in the schema validator. Now there is a request coming from some third party app that you, non-Netflix developer wrote, and request fails due to my error. As far as you are concerned, this is just an "Internal Server Error", because there is nothing you can or should do about it, other than maybe reach out to Netflix, although ideally they should be already aware of the problem before you would finish writing your email. But for that to happen it is important for this error to propagate correctly to whatever internal error handling mechanisms are implemented inside the app. So, honestly, in this case personally I wouldn't even bother to return 500 explicitly; to me it looks like some kind of FormatterError should be thrown that will be caught either by default error handler (that should return 500 by default, AFAIR), or by whatever custom one is used in the app.

Reply:
@kibertoad Thanks for the quick and very detailed response, I'll remove the handler I have created and let it be caught by the default one.

Reply:
@kibertoad There is no default handler in the `preValidationCallback`, if the handler throws, Fastify will fail, what can be done is wrapping the handler with a try catch, and returning 500: 
```js
try {
    if (reply.context.preValidation !== null) {
      hookRunner(
        reply.context.preValidation,
        hookIterator,
        request,
        reply,
        preValidationCallback
      )
    } else {
      preValidationCallback(null, request, reply)
    }
  } catch (err) {
    reply.code(500).send(err)
  }
```

If I'm missing something and there is a way to default to the regular error handler, please do tell.

Reply:
@mcollina Is there a reason why global app error handler does not apply here?